install.sh

```diff
@@ -1,63 +1,86 @@
-#!/bin/bash -e
+#!/bin/bash -ex
 set -u
 
-#The SBO can also be installed in a vanilla kubernetes cluster. A prerequisite for this would be to add credentials for the registry.redhat.io in the cluster. Steps to be followed to achieve the same are listed below:
-#1) Follow [Red Hat Container Registry Authentication Steps](https://access.redhat.com/RegistryAuthentication)
-#2) Verify that your credentials are correct using docker login -u <your_username> -p <your_passwd> registry.redhat.io
-docker login registry.redhat.io
+OPENSHIFT=${OPENSHIFT:-}
 
-#3) Start a kubernetes cluster. Any Kubernetes cluster can be used as well. The script assumes that a Kubernetes is up and running and the user has logged into it.
+if [ "$OPENSHIFT" = "true" ]; then
+  INDEX_IMAGE=${INDEX_IMAGE:-}
+  if [ -n "$INDEX_IMAGE" ]; then
+    CATSRC_NAME=${CATSRC_NAME:-$(echo -n "$INDEX_IMAGE" | sed -e 's,[/:\.],-,g')}
+  else
+    CATSRC_NAME=${CATSRC_NAME:-community-operators}
+  fi
+  CATSRC_NAMESPACE="${CATSRC_NAMESPACE:-openshift-marketplace}"
```

Comment:
I do not think it is important to know if we are running against openshift or not - instead we should detect if OLM is present or not, and install it. This can be done by performing following step:

* detect if `csv/packageserver` is available and finding out the namespace
* if available, we can detect the namespace where CatalogSources live
* detect if `openshift-operators` or `operators` namespace exit

If OLM is installed, we could detect if CatalogSource with the given index image is already installed and skip its installation if it is.

We should allow non-interactive registry logins and we need to install credentials to the cluster and add to the proper service accounts

Reply:
Right, I'll update it